### ISSUE TITLE
Cluster configuration upgrade of type StandAloneBaselineUpgradeState is already in progress and cannot be interrupted.

### ISSUE THREAD

**Issue by MathijsHofkens** [Azure/service-fabric-issues#988]
I'm trying to upgrade the configuration of my cluster. But i'm getting the following error:

```
Start-ServiceFabricClusterConfigurationUpgrade : System.Runtime.InteropServices.COMException (-2147017627)
FabricException: Cluster configuration upgrade of type StandAloneBaselineUpgradeState is already in progress and cannot be interrupted.
```

If I execute Get-ServiceFabricClusterConfigurationUpgradeStatus:
```
           UpgradeState ProgressStatus ConfigVersion Details
           ------------ -------------- ------------- -------
RollingForwardCompleted              0 1.0.0
```

What to do? Logs aren't telling me much...

**Comment by yderi** [comment]
Login to Service Fabric Explorer and have a look into Details tab, look for nodes that are not completed an update procedure or have some error

**Comment by MathijsHofkens** [comment]
Everything is green... the cluster is 3 days old and this is the first configuration update.

**Comment by yderi** [comment]
No matter how old is your cluster. For me I have got such problem right after first time I've setup my cluster.  Investigate each node for node version and look into Detail tab for each node (scroll to the end). Essential tab not always represent a real problem with node.

**Comment by rishirsinha** [comment]
@MathijsHofkens  @yderi  is right. Please go to the cluster link on the left pane. The on the right pane go to details. scroll down a bit. You will see the upgrade section. This should tell you what is happening with the upgrade. Baseline upgrade should in general finish within a few minutes. So if it is not done for 3 days, it is something we would like to investigate. If you can share traces from the cluster it would be great.

**Comment by hkuizenga** [comment]
I'm having the exact same problem.  I want to run a cluster upgrade but got the message above when trying to start it.  I ran `Get-ServiceFabricClusterUpgrade` and got the following:

![image](https://user-images.githubusercontent.com/36278411/38958027-e8ce0754-4310-11e8-9360-b120651ab743.png)

I know what I need to do to fix this UnhealthyEvaluation (ironically, that's why I'm trying to run the configuration upgrade), but I can't get past this.  Is there a way to get this to either rollback or complete even with the unhealthy evaluation?

This is a production cluster and we're stuck.

**Comment by hkuizenga** [comment]
@rakshitatandon It looks like the failure action on this is set to Manual.  Assuming I can fix the underlying issue, is it possible to get this started again via the PowerShell scripts?

**Comment by rakshitatandon** [comment]
No, you wont be able to interrupt the baseline upgrade using another upgrade via Start-ServiceFabricClusterConfigurationUpgrade. We might need to force the upgrade through using a new manifest (which has your issue fixed) and roll that out via Start-ServiceFabricClusterUpgrade. I would advise against doing this without Microsoft support.

**Comment by rakshitatandon** [comment]
One thing you could try is to fix the DCA issue (you would need to ACL it to Local System if its on the same machine as the cluster node) and then rollout the baseline upgrade again (via Start-ServiceFabricClusterUpgrade).

**Comment by hkuizenga** [comment]
That's what I was hoping :)  What parameters do I need to use with Start-ServiceFabricClusterUpgrade?

**Comment by hkuizenga** [comment]
@rakshitatandon Got the following when attempting that command:

![image](https://user-images.githubusercontent.com/36278411/38963226-b89836d0-4325-11e8-83cf-dc26032133c2.png)

**Comment by rakshitatandon** [comment]
You would have to pass in -CodePackageVersion "<SF Version used for deployment>" parameter as well and remove -Config flag.

**Comment by hkuizenga** [comment]
@rakshitatandon I ran a number of different variations w/ values for -CodePackageVersion, but still getting red letters.

![image](https://user-images.githubusercontent.com/36278411/38963487-16dfb00a-4327-11e8-913d-d138acf46213.png)

**Comment by rakshitatandon** [comment]
The CodePackageVersion needs to be the SF Version you are using (if you are using the latest then it would look something like 6.1.480.9494). Please check which version you used for your deployment.

**Comment by hkuizenga** [comment]
Got it running with that. May still need to resolve the underlying issue, but at least I know how to get the upgrade going again.  Thank you once again.

**Comment by MathijsHofkens** [comment]
@rishirsinha this is the info you requested:
![image](https://user-images.githubusercontent.com/10099914/38976149-a47d3318-43b0-11e8-9792-bfc76128f0f6.png)

How do I collect traces and upload them?

**Comment by rishirsinha** [comment]
Looks like in this case the cluster baseline upgrade did not even start. Can you send the traces in the fabriclogroot folder from the nodes where UOS primary and CM primary were?

**Comment by hkuizenga** [comment]
I was able to run `Start-ServiceFabricClusterUpgrade -CodePackageVersion 6.1.472.9494 -ClusterManifestVersion "0" -MaxPercentUnhealthyNodes 20 -Monitored -FailureAction Manual` which kicked off a cluster upgrade and ran to completion.

Unfortunately, I'm still getting the original error on this thread even though `Get-ServiceFabricClusterUpgrade` shows completed.

![image](https://user-images.githubusercontent.com/36278411/39003423-af56ccbe-43af-11e8-9b53-00d095762436.png)

**Comment by rakshitatandon** [comment]
@MathijsHofkens Please email the traces to ratando@microsoft.com.
@hkuizenga We might need to fix some cluster state here. Can you pipe the output of Get-ServiceFabricUpgradeOrchestrationServiceState to a file and send it to us ? In addition can you also send the latest clustermanifest.xml that you see in the FabricDataRoot.

**Comment by hkuizenga** [comment]
@rakshitatandon First attempt at Get-ServiceFabricUpgradeOrchestrationServiceState timed out.  I'm running it again with a 5 minute timeout but, I'm wondering how long this is expected to take?

![image](https://user-images.githubusercontent.com/36278411/39010377-59a2acac-43c3-11e8-9942-542794864649.png)

**Comment by rakshitatandon** [comment]
@hkuizenga Can you send us the cluster config (JSON) you are using instead ?

**Comment by MathijsHofkens** [comment]
I have managed to fix our cluster with the following command's:
First I’ve edited the FabricHostSvc in services.msc to run as our service account (it was running as local system)

Then I executed the following commands:
```
Connect-ServiceFabricCluster …
Get-ServiceFabricClusterManifest > ClusterManifest.xml
Copy-ServiceFabricClusterPackage -ImageStoreConnectionString fabric:ImageStore -Config -ClusterManifestPath E:\ClusterManifest.xml
Register-ServiceFabricClusterPackage -ClusterManifestPath ClusterManifest.xml -CodePackagePath MicrosoftAzureServiceFabric.6.1.480.9494.cab -TimeoutSec 600
Start-ServiceFabricClusterUpgrade -CodePackageVersion "6.1.480.9494" -ClusterManifestVersion 0 -Monitored -FailureAction Rollback –Verbose
```

**Comment by hkuizenga** [comment]
Thank you @rakshitatandon for helping us solve this!  Ultimately, we had to force data loss on the UpgradeOrchestrationService to get it unstuck.  The steps were:

```
Get-ServiceFabricClusterConfiguration | Out-File <Filepath>
Start-ServiceFabricPartitionDataLoss -OperationId <Unique GUID> -DataLossMode FullDataLoss -PartitionId 00000000-0000-0000-0000-000000006000 -ServiceName "fabric:/System/UpgradeOrchestrationService"
Start-ServiceFabricClusterConfigurationUpgrade -ClusterConfigPath <Filepath as noted in Step 1> 
```

The last step does not trigger an actual cluster update, just resets the state of the UpgradeOrchestrationService.  After applying these changes, we've been able to run cluster configuration upgrades once again.
